packages/fuels-signers/src/provider.rs

```diff
@@ -432,4 +491,352 @@ impl Provider {
             .await?;
         Ok(res.results)
     }
+
+    pub async fn get_block(&self, block_id: &str) -> Result<Option<Block>, ProviderError> {
+        self.client.block(block_id).await.map_err(Into::into)
+    }
+
+    pub async fn get_blocks(&self) -> Result<Vec<Block>, ProviderError> {
+        let res = self
+            .client
+            .blocks(PaginationRequest {
+                cursor: None,
+                results: 999,
+                direction: PageDirection::Forward,
+            })
+            .await?;
+        Ok(res.results)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    #[cfg(feature = "test-helpers")]
+    use fuel_core::model::Coin;
+
+    use fuel_gql_client::client::types::TransactionStatus;
+    use fuel_gql_client::fuel_tx::UtxoId;
+    use fuels::prelude::*;
+
+    async fn setup_provider_api_test() -> (
+        WalletUnlocked,
+        (Vec<(UtxoId, Coin)>, Vec<AssetId>),
+        Provider,
+    ) {
+        let mut wallet = WalletUnlocked::new_random(None);
+        let (coins, asset_ids) = setup_multiple_assets_coins(wallet.address(), 2, 4, 8);
+        let (provider, _) = setup_test_provider(coins.clone(), vec![], None).await;
+        wallet.set_provider(provider.clone());
+
+        (wallet, (coins, asset_ids), provider)
+    }
+
+    #[tokio::test]
+    async fn test_coin_api() -> Result<(), Error> {
+        let (_, (coins, _), provider) = setup_provider_api_test().await;
+
+        let (coin_id, _) = &coins[0];
+        let hex_coin_id = format!("{:#x}", coin_id);
+
+        let expected_coin = provider
```

Comment:
Shouldn't "expected" be the hardcoded value as opposed to the value retrieved from the tested method?

Reply:
Yes, you are right :smile:  I will update the tests.